RecoVertex/V0Producer/src/V0Fitter.cc

```diff
@@ -154,9 +154,14 @@ void V0Fitter::fitAll(const edm::Event& iEvent, const edm::EventSetup& iSetup,
       }
 
       // measure distance between tracks at their closest approach
-      if (!posTransTkPtr->impactPointTSCP().isValid() || !negTransTkPtr->impactPointTSCP().isValid()) continue;
-      FreeTrajectoryState const & posState = posTransTkPtr->impactPointTSCP().theState();
-      FreeTrajectoryState const & negState = negTransTkPtr->impactPointTSCP().theState();
+
+      //these two variables are needed to 'pin' the temporary value returned to the stack
+      // in order to keep posState and negState from pointing to destructed objects
+      auto const& posImpact = posTransTkPtr->impactPointTSCP();
+      auto const& negImpact = negTransTkPtr->impactPointTSCP();
+      if (!posImpact.isValid() || !negImpact.isValid()) continue;
+      FreeTrajectoryState const & posState = posImpact.theState();
+      FreeTrajectoryState const & negState = negImpact.theState();
```

Comment:
this is semantically more minimal in terms of copies.

Reply:
if you grep for "impactPointTSCP" you will find the same style allovertheplace
(and is not minimal only in terms of copy also in terms of virtual call and actual computation...)

Reply:
@VinInn 
my comment was in context of this scope and was comparing to #21172 .
Does your comment change something in this context? I may be missing something otherwise.

Reply:
No, not in this scope (even if I am not sure ```auto const&``` is correct and should not be
```auto const```.
I was only saying that there are many other places where isValid si first tested on the fly and then impactPointTSCP accesses again...
see
https://cmssdt.cern.ch/dxr/CMSSW/source/L1Trigger/L1TNtuples/plugins/L1MuonRecoTreeProducer.cc#829
fro instance

Reply:
@Dr15Jones could you also look/fix lines 176-177 in your patch? When I checked DXR `trajectoryStateClosestToPoint` also returns a copy: https://github.com/cms-sw/cmssw/blob/master/TrackingTools/TransientTrack/interface/BasicTransientTrack.h#L51

Reply:
@Dr15Jones ignore my comment above, const extends temporary lifetime in those lines.